tests/PHPStan/Analyser/data/unset-conditional-expressions.php

```diff
@@ -27,4 +27,22 @@ public function doFoo(array $filteredParameters, array $a): void
 		assertType('array<string, mixed>', $filteredParameters);
 	}
 
+	public function doBaz(): void
+	{
+		$breakdowns = [
+			'a' => (bool) rand(0, 1),
+			'b' => (string) rand(0, 1),
+			'c' => rand(-1, 1),
+			'd' => rand(0, 1),
+		];
+
+		foreach ($breakdowns as $type => $bd) {
+			if (empty($bd)) {
+				unset($breakdowns[$type]);
+			}
+		}
+
+		assertType('array{}|array{a?: bool, b?: numeric-string, c?: int<-1, 1>, d?: int<0, 1>}', $breakdowns);
```

Comment:
This was `array<'a'|'b'|'c'|'d', bool|int<-1, 1>|numeric-string>` before. See https://phpstan.org/r/6c90923e-abf5-492d-8bf6-c5f95a6627e6
I added it because something very similar to this was failing in prestashop while developing. It has been unblocked/fixed by Ondrej's recent changes, but I want to avoid it failing again. And it is a nice example I think.

The union with an empty array could be optimised away. If this isn't already done in the tagged unions branch anyway.. :)

Reply:
I guess in the future this could even be 
`assertType('array{}|array{a?: true, b?: numeric-string~0, c?: -1|1>, d?: 1}', $breakdowns);`
?